android/src/main/java/ie/gov/tracing/Tracing.kt

```diff
@@ -395,10 +397,9 @@ class Tracing {
             var versionName: String
             var versionCode: String
             try {
-                versionName = context.getPackageManager()
-                        .getPackageInfo(context.getPackageName(), 0).versionName
-                versionCode = context.getPackageManager()
-                        .getPackageInfo(context.getPackageName(), 0).longVersionCode.toString()
+                val pinfo: PackageInfo = context.getPackageManager().getPackageInfo(context.getPackageName(), 0)
+                versionName = pinfo.versionName
+                versionCode = PackageInfoCompat.getLongVersionCode(pinfo).toString()           
```

Comment:
Looking at the api I saw: https://developer.android.com/reference/android/content/pm/PackageInfo#getLongVersionCode(). They're doind the same thing?

Reply:
getLongVersiobCode in PackageInfo is not supported on. older devices. PackageInfoCompat solves this